addOns/authhelper/src/test/java/org/zaproxy/addon/authhelper/HeaderBasedSessionManagementMethodTypeUnitTest.java

```diff
@@ -389,4 +392,110 @@ void shouldParseHeaderValuesProperly(String entry, String expectedFirst, String
         assertThat(header.first, is(equalTo(expectedFirst)));
         assertThat(header.second, is(equalTo(expectedSecond)));
     }
+
+    @ParameterizedTest
+    @CsvSource(value = {"'',0", "'   \\t\\n ',0", "Header1:Value1,1"})
+    void shouldSetHeaderConfigsFromApi(String headers, int expectedSize) throws ApiException {
+        // Given
+        HeaderBasedSessionManagementMethodType type = new HeaderBasedSessionManagementMethodType();
+        Context context = mock(Context.class);
+        JSONObject params = new JSONObject();
+        params.put("contextId", 1);
+        params.put("headers", headers);
+
+        Model model = mock(Model.class, withSettings().strictness(Strictness.LENIENT));
+        Model.setSingletonForTesting(model);
+        Session session = mock(Session.class);
+        given(model.getSession()).willReturn(session);
+        given(session.getContext(1)).willReturn(context);
+
+        // When
+        type.getSetMethodForContextApiAction().handleAction(params);
+
+        // Then
+        ArgumentCaptor<SessionManagementMethod> captor =
+                ArgumentCaptor.forClass(SessionManagementMethod.class);
+        verify(context).setSessionManagementMethod(captor.capture());
+        HeaderBasedSessionManagementMethod savedMethod =
+                (HeaderBasedSessionManagementMethod) captor.getValue();
+        assertThat(savedMethod.getHeaderConfigs(), hasSize(expectedSize));
+    }
+
+    @Test
+    void shouldSetHeaderConfigsFromApiWhenParamMissing() throws ApiException {
+        // Given
+        HeaderBasedSessionManagementMethodType type = new HeaderBasedSessionManagementMethodType();
+        Context context = mock(Context.class);
+        JSONObject params = new JSONObject();
+        params.put("contextId", 1);
+
+        Model model = mock(Model.class, withSettings().strictness(Strictness.LENIENT));
+        Model.setSingletonForTesting(model);
+        Session session = mock(Session.class);
+        given(model.getSession()).willReturn(session);
+        given(session.getContext(1)).willReturn(context);
+
+        // When
+        type.getSetMethodForContextApiAction().handleAction(params);
+
+        // Then
+        ArgumentCaptor<SessionManagementMethod> captor =
+                ArgumentCaptor.forClass(SessionManagementMethod.class);
+        verify(context).setSessionManagementMethod(captor.capture());
+        HeaderBasedSessionManagementMethod savedMethod =
+                (HeaderBasedSessionManagementMethod) captor.getValue();
+        assertThat(savedMethod.getHeaderConfigs(), hasSize(0));
+    }
+
+    @Test
+    void shouldLeaveUnresolvedJsonTokenInHeader() throws Exception {
```

Comment:
Why these tests unrelated to the fix and already covered by `shouldExtractWebSession`?